dvc/config_schema.py

```diff
@@ -179,7 +179,12 @@ class RelPath(str):
                     "webhdfs_alias": str,
                     **REMOTE_COMMON,
                 },
-                "azure": {"connection_string": str, **REMOTE_COMMON},
+                "azure": {
+                    "connection_string": str,
+                    "sas_token": str,
+                    "account_name": str,
```

Comment:
I see that it's already not only about  the migration, meaning that we need to address principal credentials, ideally other env vars Azure lib supports as we have been discussing in the second Azure PR

Reply:
agreed

Reply:
These are the environment variables that adlfs fallbacks to (unless we give it from our config / local azure config):
https://github.com/dask/adlfs/blob/f7e810ab5294037f0d7ad5db16e480e119a48c3f/adlfs/spec.py#L375-L383

Maybe we could just document those?

Reply:
Actually we already handle it in our `_az_config` section (knack fallbacks to reading environment variables with `AZURE_{SECTION}_{KEY}` format;
https://github.com/iterative/dvc/blob/53ba51b5fed1ed19f03c3050ccd83ca128ce9012/dvc/fs/azure.py#L57-L67

though we don't comply with Azure docs there (this is also the old behavior, not something new);
https://docs.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python#service-principal-with-secret